core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -713,8 +718,15 @@ private[spark] class BlockManager(
   /**
    * Release a lock on the given block.
    */
-  def releaseLock(blockId: BlockId): Unit = {
-    blockInfoManager.unlock(blockId)
+  def releaseLock(blockId: BlockId): Unit = releaseLock(blockId, taskAttemptId = None)
```

Comment:
In fact `BlockManager` extends `BlockDataManager`, so it have to override the `releaseLock(blockId: BlockId)` method, thus we keep this and implement a new method that accepts the new argument `taskAttemptId`.